scaladoc/src/dotty/tools/scaladoc/tasty/Scaladoc2AnchorCreator.scala

```diff
@@ -7,10 +7,66 @@ import scala.util.matching.Regex
 
 object Scaladoc2AnchorCreator:
 
-  def getScaladoc2Type(using Quotes)(t: reflect.Tree) =
-    import reflect.*
-    val regex = t match
-      case d: DefDef =>  "def"
-      case t: TypeDef => "type"
-      case v: ValDef =>  "val|var"
-    t.show(using Printer.TreeShortCode).split(regex, 2)(1).replace(" ","")
+  def getScaladoc2Type(using Quotes)(sym: quotes.reflect.Symbol) = signatureAnchor(sym)
+
+  /** Creates the signature anchor
+   *
+   *  - `X` for a `type X ...`
+   *  - `x:X` for a `val x: X`
+   *  - `f[U1,...](x1:T1,...)(impliciti1:U1,impliciti2:U2,...)...:R` for a `def f[U1, ...](x1: T1, ...)(implicit i1: U1, i2: U2...)...: R`
+   *
+   *  Types are printed without their paths. No spaces are printed in the output.
+   */
+  private def signatureAnchor(using Quotes)(sym: quotes.reflect.Symbol): String =
+    import quotes.reflect.*
+    def signatureType(tp: quotes.reflect.TypeRepr): String =
+      tp match
+        case mt @ MethodType(paramNames, paramTypes, res) =>
+          val implicitPrefix = if mt.isImplicit then "implicit " else ""
```

Comment:
There should be no space after `implicit` here. Otherwise LGTM

Reply:
Oops